samples/subsys/ipc/openamp_rsc_table/src/main_remote.c

```diff
@@ -53,26 +54,17 @@ static const struct device *const ipm_handle =
 
 static metal_phys_addr_t shm_physmap = SHM_START_ADDR;
 
-struct metal_device shm_device = {
-	.name = SHM_DEVICE_NAME,
-	.num_regions = 2,
-	.regions = {
-		{.virt = NULL}, /* shared memory */
-		{.virt = NULL}, /* rsc_table memory */
-	},
-	.node = { NULL },
-	.irq_num = 0,
-	.irq_info = NULL
-};
+static struct metal_io_region shm_io_data; /* shared memory */
+static struct metal_io_region rsc_io_data; /* rsc_table memory */
```

Comment:
what about naming them `shm_io` and `rsc_io ` and remove ` static struct metal_io_region *shm_io ` and `static struct metal_io_region *shm_io`?

Reply:
I thought about that originally, but this causes less churn at each point they are used (removing all the dereferences). And at some point in the future I want to make `metal_io_region` an opaque struct. So these would need to become a pointer again, so less churn changing them back too.